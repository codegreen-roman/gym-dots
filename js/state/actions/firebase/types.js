export const GOT_DEFAULTS = 'GOT_DEFAULTS'
export const GOT_ERROR_LOADING_DEFAULTS = 'GOT_ERROR_GETTING_DEFAULTS'
export const START_AUTH = 'START_AUTH'
export const COMPLETE_AUTH = 'COMPLETE_AUTH'
export const AUTH_SUCCESS = 'AUTH_SUCCESS'
export const AUTH_ERROR = 'AUTH_ERROR'
export const EXERCISES_FETCHING_SUCCESS = ' EXERCISES_FETCHING_SUCCESS'
export const EXERCISES_FETCHING_ERROR = ' EXERCISES_FETCHING_ERROR'
