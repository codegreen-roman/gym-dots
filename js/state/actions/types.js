export const TEST_ACTION = 'TEST_ACTION'
export const GET_LOCATION = 'GET_LOCATION'
export const GET_LOCATION_START = 'GET_LOCATION_START'
export const PING = 'PING'
export const LOAD_SESSIONS_SUCCESS = 'LOAD_SESSIONS_SUCCESS'

