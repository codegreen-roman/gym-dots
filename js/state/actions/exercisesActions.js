import { EXERCISES_ORDER_CHANGE, EXERCISES_MOVE_EXERCISE_TO_COMPLETED } from './types'

export const exercisesOrderChange = (exercise) => ({
    type: EXERCISES_ORDER_CHANGE,
    payload: {
        exercise
    }
})

export const moveExerciseToCompleted = (exerciseKey) => ({
    type: EXERCISES_MOVE_EXERCISE_TO_COMPLETED,
    payload: {
        exerciseKey
    }
})
