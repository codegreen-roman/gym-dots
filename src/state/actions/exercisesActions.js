import {
    EXERCISES_ORDER_CHANGE,
    EXERCISES_MOVE_EXERCISE_TO_COMPLETED,
    EXERCISES_SAVED_RESULTS_SUCCESS,
    EXERCISES_SAVED_RESULTS_FAILED
} from './types'

export const exercisesOrderChange = (exercise) => ({
    type: EXERCISES_ORDER_CHANGE,
    payload: {
        exercise
    }
})

export const moveExerciseToCompleted = (exerciseKey, results) => ({
    type: EXERCISES_MOVE_EXERCISE_TO_COMPLETED,
    payload: {
        exerciseKey,
        results
    }
})

export const saveResultsCompleted = (data) => ({
    type: EXERCISES_SAVED_RESULTS_SUCCESS,
    payload: {
        data
    }
})

export const saveResultsFailed = () => ({
    type: EXERCISES_SAVED_RESULTS_FAILED,
    payload: {
    }
})
