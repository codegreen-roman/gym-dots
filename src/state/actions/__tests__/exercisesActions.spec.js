import { exercisesOrderChange } from '../exercisesActions'
import { EXERCISES_ORDER_CHANGE } from '../types'

describe('Exercises Action Creators', () => {
    describe('.exercisesOrderChange', () => {
        it('should create EXERCISES_ORDER_CHANGE action', () => {
            const expectedAction = {
                type: EXERCISES_ORDER_CHANGE,
                payload: {
                    exercise: {
                        exerciseId: '3',
                        restTime: 90,
                        name: 'Push-ups',
                        sets: 5,
                        reps: 20,
                        weight: 0,
                        results: []
                    }
                }
            }
            expect(
                exercisesOrderChange({
                    exerciseId: '3',
                    restTime: 90,
                    name: 'Push-ups',
                    sets: 5,
                    reps: 20,
                    weight: 0,
                    results: []
                })
            ).toEqual(expectedAction)
        })

        it('should return action with a type of EXERCISES_ORDER_CHANGE', () => {
            expect(
                exercisesOrderChange({
                    exerciseId: '3',
                    restTime: 90,
                    name: 'Push-ups',
                    sets: 5,
                    reps: 20,
                    weight: 0,
                    results: []
                })
            ).toMatchSnapshot()
        })
    })
})
