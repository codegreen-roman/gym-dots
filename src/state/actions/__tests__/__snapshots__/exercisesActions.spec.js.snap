// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercises Action Creators .exercisesOrderChange should return action with a type of EXERCISES_ORDER_CHANGE 1`] = `
Object {
  "payload": Object {
    "exercise": Object {
      "exerciseId": "3",
      "name": "Push-ups",
      "reps": 20,
      "restTime": 90,
      "results": Array [],
      "sets": 5,
      "weight": 0,
    },
  },
  "type": "EXERCISES_ORDER_CHANGE",
}
`;
